spec/features/consumer/shopping/embedded_groups_spec.rb

```diff
@@ -63,16 +53,29 @@
       end
     end
 
-    it 'opens links to shops in a new window' do
-      expect(page).to have_selector 'iframe#group_test_iframe'
-
-      within_frame 'group_test_iframe' do
+    it "opens links to shops in a new window" do
+      on_embedded_page do
         within 'div#group-page' do
-          enterprise_links = page.all(:xpath, "//*[contains(@href, 'enterprise-5/shop')]", :visible => :false).count
-          enterprise_links_with_target_blank = page.all(:xpath, "//*[contains(@href, 'enterprise-5/shop') and @target = '_blank']", :visible => :false).count
-          expect(enterprise_links).to equal(enterprise_links_with_target_blank)
+          shop_links_xpath = "//*[contains(@href, '#{enterprise.permalink}/shop')]"
+
+          expect(page).to have_xpath shop_links_xpath, visible: false
+
+          shop_links = page.all(:xpath, shop_links_xpath, visible: false)
+          shop_links.each do |link|
+            expect(link[:target]).to eq "_blank"
+          end
         end
       end
     end
   end
+  
+  private
+  
+  def on_embedded_page
+      expect(page).to have_selector "iframe"
```

Comment:
do we really need this expectation here? the tests that use this method already check other assertions so I'm not sure it's worth the cost of this extra `expect`.